kmath-gsl/src/nativeMain/kotlin/kscience/kmath/gsl/GslRealMatrix.kt

```diff
@@ -0,0 +1,47 @@
+package kscience.kmath.gsl
+
+import kotlinx.cinterop.CStructVar
+import kotlinx.cinterop.CValues
+import kotlinx.cinterop.memScoped
+import kotlinx.cinterop.pointed
+import kscience.kmath.linear.FeaturedMatrix
+import kscience.kmath.linear.MatrixFeature
+import kscience.kmath.structures.NDStructure
+import org.gnu.gsl.gsl_matrix
+import org.gnu.gsl.gsl_matrix_equal
+import org.gnu.gsl.gsl_matrix_get
+
+public sealed class GslMatrix<T : Any> : FeaturedMatrix<T> {
+    public abstract val nativeHandle: CValues<out CStructVar>
+
+    public override fun hashCode(): Int {
+        var result = nativeHandle.hashCode()
+        result = 31 * result + features.hashCode()
+        return result
+    }
+}
+
+public class GslRealMatrix(public override val nativeHandle: CValues<gsl_matrix>, features: Set<MatrixFeature>) :
```

Comment:
handle should probably be private

Reply:
That's true, but the current challenge is to build this at least

Reply:
Currently, I can't even run cinterop in Actions